### PR TITLE
Add update for pd client if tls config changes

### PR DESCRIPTION
**Pull request by birdstorm**
Add update API for PD client so that it can refresh configurations when TLS config changes.

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

**Review by LittleFall — APPROVED** [review]

**Review by SeaRise — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]

File: src/pd/Client.cc [hunk 1]
```diff
@@ -63,6 +63,31 @@ Client::~Client()
...
+    urls = addrsToUrls(addrs, config_);
+    work_threads_stop = false;
+    check_leader = false;
+    config = config_;
+
+    initClusterID();
+
+    initLeader();
+
+    work_threads_stop = false;
```
Comment (gengliqi) [inline src/pd/Client.cc:84]: Why does `work_threads_stop` set twice?

Comment (ywqzzy) [inline src/pd/Client.cc:84]:
```suggestion
```

Comment (birdstorm) [reply src/pd/Client.cc:84]: done

File: src/pd/Client.cc [hunk 2]
```diff
@@ -42,25 +56,33 @@ Client::Client(const std::vector<std::string> & addrs, const ClusterConfig & con
...
 
-Client::~Client()
+void Client::update(const std::vector<std::string> & addrs, const ClusterConfig & config_)
 {
     work_threads_stop = true;
 
     if (work_thread.joinable())
     {
         work_thread.join();
     }
```
Comment (gengliqi) [inline src/pd/Client.cc:79]: How about adding a `Uninit`?

File: src/pd/Client.cc [hunk 3]
```diff
@@ -42,25 +56,33 @@ Client::Client(const std::vector<std::string> & addrs, const ClusterConfig & con
...
+
+    urls = addrsToUrls(addrs, config_);
+    check_leader = false;
+    config = config_;
```
Comment (gengliqi) [inline src/pd/Client.cc:83]: Seems `urls` and `config` can also be moved to `init`. `check_leader` has already existed in `init`.

File: src/pd/Client.cc [hunk 4]
```diff
@@ -30,18 +30,12 @@ inline std::vector<std::string> addrsToUrls(const std::vector<std::string> & add
...
-    , urls(addrsToUrls(addrs, config_))
-    , cluster_id(0)
-    , work_threads_stop(false)
-    , check_leader(false)
-    , config(config_)
-    , log(&Logger::get("pingcap.pd"))
+void Client::init(const std::vector<std::string> & addrs, const ClusterConfig & config_)
 {
+    urls = addrsToUrls(addrs, config_);
+    check_leader = false;
```
Comment (gengliqi) [inline src/pd/Client.cc:36]: This line is the same as line 47.

File: src/pd/Client.cc [hunk 5]
```diff
@@ -63,6 +57,31 @@ Client::~Client()
     }
 }
 
+Client::Client(const std::vector<std::string> & addrs, const ClusterConfig & config_)
+    : max_init_cluster_retries(100)
+    , pd_timeout(3)
+    , loop_interval(100)
+    , update_leader_interval(60)
+    , cluster_id(0)
+    , work_threads_stop(false)
```
Comment (gengliqi) [inline src/pd/Client.cc:66]: `work_threads_stop` is set in `init`.

Comment (birdstorm) [reply src/pd/Client.cc]: fixed

Comment (gengliqi) [reply src/pd/Client.cc]: The same to `check_leader`.

**Pull request merged**